mbuild/utils/specific_ff_to_residue.py

```diff
@@ -0,0 +1,326 @@
+                           forcefield_selection=None,
+                           residues=None,
+                           reorder_res_in_pdb_psf=False,
+                           box=None,
+                           boxes_for_simulation=1):
+    """ToDo: List What this function does??
+    Parameters
+    ----------
+    structure: mb.Compound
+        The mBuild Compound object
+    forcefield_selection: str or dictionary, default=None
+        Apply a forcefield to the output file by selecting a force field xml file with
+        its path or by using the standard force field name provided the `foyer` package.
+        See Notes for further details on this argument
+    residues: str of list of str, default=None
+        Labels of unique residues in the Compound. Residues are assigned by
+        checking against Compound.name.  Only supply residue names as 3 characters
+        strings, as the residue names are truncated to 3 characters to fit in the
+        psf and pdb file.
+    reorder_res_in_pdb_psf: bool, default=False
+        ToDo: Add Description fot this argument
+    box: list, default=None
+        list of 3 positive float values or the dimensions [x, y ,z]
+        for structure in nanometers (nm). This is to add/override or change the structures dimensions.
+        Ex: [1,2,3]
+    boxes_for_simulation: {1, 2}
+        Gibbs or grand canonical ensembles are examples of where the boxes_for_simulation would be 2
+
+    Notes
+    -----
+    To write the NAMD/GOMC force field, pdb, and psf files, the
+    residues and forcefields must be provided in a str or
+    dictionary.  If a dictionary is provided all residues must
+    be specified to a force field.
+
+    Example dict for FF file: {'ETH' : 'oplsaa.xml', 'OCT': 'path_to file/trappe-ua.xml'}
+    Example str for FF file: 'path_to file/trappe-ua.xml'
+    Example dict for standard FF names : {'ETH' : 'oplsaa', 'OCT': 'trappe-ua'}
+    Example str for standard FF names: 'trappe-ua'
+    Example of a mixed dict with both : {'ETH' : 'oplsaa', 'OCT': 'path_to file/'trappe-ua.xml'}
+
+    Returns
+    -------
+    structure: parmed.Structure
+        parmed structure with applied force field
+    coulomb14scaler_dict: dict
+        a dictionary with the 1,4-colombic scalers for each residue
+            (i.e., a different force field could on each residue)
+    lj14_scaler_dict: dict
+        a dictionary with the 1,4-LJ scalers for each residue
+        (i.e., a different force field could on each residue)
+    residues_applied_list: list
+        list of residues (i.e., list of stings).
+        These are all the residues in which the force field actually applied
+    """
+
+    if has_foyer:
+        from foyer import Forcefield
+        from foyer.forcefields import forcefields
+    else:
+        warn(
+            'Package foyer is not installed. '
+            'Please install it using conda install -c conda-forge foyer'
+        )
+        return None, None, None, None
```

Comment:
We can return an error.  I thought this as more of a function.  I was debating myself whether I should return an error.  I can return as a list.  I would have to modify a bit but it can be done.  Further discussion on this would be helpful for me.  

Basically the None value here shutdown the rest of the analysis if the are requested by the Charmm writer and return None

Reply:
This file returns errors now and not None's.  Also, it return as a list